### DISCUSSION TITLE
Error with importing Counter from machine

### DISCUSSION THREAD

**Discussion by JonathanRNL** [micropython/micropython#9486] [Using MicroPython] [answered]
Hello all,

I have a problem with importing the counter from the machine library.

I use the Thonny IDE for programming a LilyGO TTGO T3 LoRa32 868MHz V1.6.1 ESP32 Microcontroller Board in MicroPython. I connect the device via a MicroUSB cable. I use MicroPython v1.14.

When I run the following code:

---
    py
    from machine import Counter
---

I will get the following message:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: can't import name Counter

I tried it on two independent PCs. Unfortunately, it did not work on either one of them.

Any idea how I can solve this problem?

Kind regards,

Jonathan

**Comment by robert-hh** [comment, marked as answer]
The Counter class is not included in the machine module yet. There exists various PRs for that, but none of them is merged. If you want to use is, you have to build your own firmware. Open PRs:
- #8766
- #7582 with #5216
- #8253

**Reply by JonathanRNL** [reply]
Thank you. Is there not anybody somewhere that already merged everything?

**Reply by robert-hh** [reply]
I can try to build a firmware image with Counter for you.

**Reply by robert-hh** [reply]
There is a bug in the build system, which prevents building of that PR at the moment. Should be fixed soon.

**Reply by robert-hh** [reply]
Try the firmare binary here; https://github.com/robert-hh/Shared-Stuff/blob/master/firmware_LILYGO_TTGO_LORA32.bin
 It is based on PR #8766, which is documented in PR #8072.

**Reply by JonathanRNL** [reply]
> Try the firmare binary here; https://github.com/robert-hh/Shared-Stuff/blob/master/firmware_LILYGO_TTGO_LORA32.bin It is based on PR #8766, which is documented in PR #8072.

Thank you very much! The problem is solved now :).
